test/powershell/Language/Operators/NullConditional.Tests.ps1

```diff
@@ -264,3 +264,114 @@ Describe 'NullConditionalOperations' -Tags 'CI' {
         }
     }
 }
+
+Describe 'NullConditionalMemberAccess' -Tag 'CI' {
+
+    BeforeAll {
+        $skipTest = -not $EnabledExperimentalFeatures.Contains('PSNullConditionalOperators')
+
+        if ($skipTest) {
+            Write-Verbose "Test Suite Skipped. The test suite requires the experimental feature 'PSNullConditionalOperators' to be enabled." -Verbose
+            $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+            $PSDefaultParameterValues["it:skip"] = $true
+        }
+    }
+
+    AfterAll {
+        if ($skipTest) {
+            $global:PSDefaultParameterValues = $originalDefaultParameterValues
+        }
+    }
+
+    Context '?. operator tests' {
+        BeforeAll {
+            $psObj = [psobject]::new()
+            $psObj | Add-Member -Name 'name' -Value 'value' -MemberType NoteProperty
+            $psObj | Add-Member -Name 'nested' -Value @{name = 'valuenested'} -MemberType NoteProperty
+
+            $psobj2 = [psobject]::new()
+            $psobj2 | Add-Member -Name 'GetHello' -Value { "hello" } -MemberType ScriptMethod
+            $psObj | Add-Member -Name 'nestedMethod' -Value $psobj2 -MemberType NoteProperty
+
+            $array = 1..3
+            $hash = @{ a = 1; b = 2}
+
+            $null = New-Item -ItemType File -Path "$TestDrive/testfile.txt" -Force
+        }
+
+        It 'Can get member value of a non-null variable' {
+            ${psObj}?.name | Should -BeExactly 'value'
+            ${array}?.length | Should -Be 3
+            ${hash}?.a | Should -Be 1
+
+            (Get-Process -Id $pid)?.Name | Should -BeLike "pwsh*"
+            (Get-Item $TestDrive)?.EnumerateFiles()?.Name | Should -BeExactly 'testfile.txt'
+
+            [int32]::MaxValue?.ToString() | Should -BeExactly '2147483647'
+        }
+
+        It 'Can get null when variable is null' {
+            ${nonExistent}?.name | Should -BeNullOrEmpty
+            ${nonExistent}?.MyMethod() | Should -BeNullOrEmpty
+
+            (get-process -Name doesnotexist -ErrorAction SilentlyContinue)?.Id | Should -BeNullOrEmpty
+        }
+
+        It 'Use ?. operator multiple times in statement' {
+            ${psObj}?.name?.nonExistent | Should -BeNullOrEmpty
+            ${psObj}?.nonExistent?.nonExistent | Should -BeNullOrEmpty
+            ${nonExistent}?.nonExistent?.nonExistent | Should -BeNullOrEmpty
+
+            ${psObj}?.nested?.name | Should -BeExactly 'valuenested'
+            ${psObj}?.nestedMethod?.GetHello() | Should -BeExactly 'hello'
+        }
+
+        it 'Use ?. on a dynamic method name' {
+            $methodName = 'ToLongDateString'
+            (Get-Date '11/11/2019').$methodName()
+        }
```

Comment:
1. this is not used with `?.`.
2. there is no check on the result

Maybe add a few more test cases to verify the dynamic member name scenario, including member and method, with the target to be null or non-null.

Reply:
Added more test cases.